vitest-setup.ts

```diff
@@ -1,17 +1,31 @@
 import { vi } from "vitest";
+import { EnvironmentMode } from "./src/config";
 
-vi.mock("./src/config", async () => ({
-  AWS_REGION: "ca-central-1",
-  SERVER_PORT: 3001,
-  LOCALSTACK_ENDPOINT: undefined,
-  ZITADEL_APPLICATION_KEY: JSON.stringify({
-    keyId: "test",
-    clientId: "test",
-    key: "test",
-  }),
-  ZITADEL_DOMAIN: "test",
-}));
+vi.doMock("./src/config", async (importOriginal) => {
+  const original = (await importOriginal()) as object;
 
-vi.mock("node:crypto", () => ({
-  createPrivateKey: vi.fn(),
-}));
+  return {
+    ...original,
+    AWS_REGION: "ca-central-1",
+    ENVIRONMENT_MODE: EnvironmentMode.Production,
```

Comment:
Set this to `Production` to make sure the code we are testing is the one that will be running in production.

Reply:
This will, for instance, allow the `FreshdeskApiClient` to actually call `Axios` to make a request. This is because, in `Local` mode, the `createTicket` function just responds early with a successful `Promise`.

Reply:
If it starts being complex to write unit tests because of that we can always change it and have the function relying on the `EnvironmentMode` variable to do something just has to request it in its inputs.

Reply:
Will change it because it is better in terms of clean code to have the function request the Env Mode as an input.